### PR TITLE
dynparquet: add String method to SerializedBuffer

### PR DESCRIPTION
**Pull request by asubiotto**
Address #331. The result looks something like:
```
row groups: 1  total rows: 9
---
[name]           [sample_type]  [sample_unit]  [period_type]  [period_unit]  [labels.agen...  [labels.app_...  [labels.app_...  [labels.app_...  [labels.app_...  [labels.app_...  [labels.app_...  [labels.cgro...  [labels.comm]    [labels.cont...  [labels.cont...  [labels.cont...  [labels.exec...  [labels.k8s_...  [labels.kern...  [labels.name...  [labels.node]    [labels.oper...  [labels.oper...  [labels.pid]   [labels.pod]     [labels.pod_...  [labels.ppid]  [labels.prom...  [labels.stat...  [labels.syst...  [timestamp]
parca_agent_cpu  samples        count          cpu            nanoseconds    0b3a3e99501c...  prometheus       k8s              prometheus-o...  prometheus       kube-prometheus  2.39.1           /kubepods/bu...  prometheus       prometheus       containerd:/...  prometheus-k...  /bin/prometheus                   5.10.147+        observability    gke-europe-w...  k8s              0                4921           prometheus-k...                   4722           k8s              prometheus-k...                   1675794285440
parca_agent_cpu  samples        count          cpu            nanoseconds    0b3a3e99501c...                   loki                              promtail                                           /kubepods/be...  promtail         promtail         containerd:/...  586b8bb487       /usr/bin/pro...                   5.10.147+        observability    gke-europe-w...                                    41567          loki-promtai...  1                41489                                                             1675794285440
parca_agent_cpu  samples        count          cpu            nanoseconds    0b3a3e99501c...                                                                                                        /kubepods/bu...  cilium-agent     cilium-agent     containerd:/...  69bf5b5bc6       /usr/bin/cil...  cilium           5.10.147+        kube-system      gke-europe-w...                                    2706           anetd-6464b      24               2048                                                              1675794285440
parca_agent_cpu  samples        count          cpu            nanoseconds    0b3a3e99501c...                                                                                                        /system.slic...  containerd                                                          /usr/bin/con...                   5.10.147+                         gke-europe-w...                                    1621                                             1                                                containerd.s...  1675794285440
parca_agent_cpu  samples        count          cpu            nanoseconds    0b3a3e99501c...                                                                                                        /system.slic...  containerd-shim                                                     /usr/bin/con...                   5.10.147+                         gke-europe-w...                                    3972                                             1                                                containerd.s...  1675794285440
parca_agent_cpu  samples        count          cpu            nanoseconds    0b3a3e99501c...                                                                                                                                                                                                                               5.10.147+                         gke-europe-w...                                    65671                                                                                                              1675794285440
parca_agent_cpu  samples        count          cpu            nanoseconds    0b3a3e99501c...                                                                                                                                                                                                                               5.10.147+                         gke-europe-w...                                    65687                                                                                                              1675794285440
parca_agent_cpu  samples        count          cpu            nanoseconds    0b3a3e99501c...                                                                                                        /system.slic...  kubelet                                                             /home/kubern...                   5.10.147+                         gke-europe-w...                                    1801                                             1                                                kubelet.service  1675794285440
parca_agent_cpu  samples        count          cpu            nanoseconds    0b3a3e99501c...                                                                                                                                                                                                                               5.10.147+                         gke-europe-w...                                    65665                                                                                                              1675794285440
```

### CODE REVIEW

**Comment by asubiotto** [comment]
Only the sorting columns are printed out since I think including all columns would be confusing

File: dynparquet/reader.go [hunk 1]
```diff
@@ -72,6 +73,63 @@ func (b *SerializedBuffer) DynamicRows() DynamicRowReader {
...
+		}
+		return s
+	}
+
+	numRowGroups := b.NumRowGroups()
+	numRows := b.NumRows()
+	_, _ = w.Write([]byte(fmt.Sprintf("row groups: %d\ttotal rows: %d\n", numRowGroups, numRows)))
+	for i := 0; i < numRowGroups; i++ {
+		_, _ = w.Write([]byte("---\n"))
+		func() {
```
Comment (metalmatze) [inline dynparquet/reader.go:100]: Any specific reason for this `func()`? Is it for the `defer rows.Close()`?

**Review by metalmatze — COMMENTED** [review]
Very cool!

File: dynparquet/reader.go (hunk 1 above)
Comment (asubiotto) [reply dynparquet/reader.go:100]: yes

File: dynparquet/reader.go [hunk 2]
```diff
@@ -72,6 +73,63 @@ func (b *SerializedBuffer) DynamicRows() DynamicRowReader {
 	return Concat(b.fields, drg...).DynamicRows()
 }
 
+func (b *SerializedBuffer) String() string {
```
[the remark this answers is not in the document]
Comment (asubiotto) [reply dynparquet/reader.go:76]: yes, it's the `fmt.Stringer` interface which is recognized by go and used whenever you print the object.

**Review by metalmatze — APPROVED** [review]

**Pull request merged**